app/features/workout/WorkoutList.js

```diff
@@ -36,14 +37,28 @@ class WorkoutList extends Component {
     // RENDER
 
     _renderSectionHeader(section) {
+        let sets = this.props.sets;
+        let currentSetIndex = (this.props.sets.length) - 1;
+        let set = sets[currentSetIndex];
+
         if (section.key === 0) {
-            return (
-                <View style={styles.button}>
-                    <TouchableOpacity onPress={ () => this.props.endSet() }>
-                        <Text style={styles.buttonText}>Add New Set</Text>
-                    </TouchableOpacity>
-                </View>
-            );
+            if (!set.exercise && !set.weight && !set.rpe && (set.reps !== null && set.reps !== undefined && set.reps.length === 0) && (set.tags !== null && set.tags !== undefined && set.tags.length === 0) && !set.videoFileURL) {
```

Comment:
Move this if statement logic to its own function or add a comment explaining what the if statement is doing